### PR TITLE
Disable metrics service

### PR DESCRIPTION
**Pull request by EItanya**
# Description

This PR removes the currently unused, and CPU intensive metrics service

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo [contribution guide](https://docs.solo.io/gloo/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3849

### CODE REVIEW

File: .github/workflows/publish-docs.yaml [hunk 1]
```diff
@@ -33,4 +33,5 @@ jobs:
       env:
         PROJECT_ID: solo-corp
         PROJECT_PATH: ./docs/ci
-        GCP_SA_KEY: ${{ secrets.FIREBASE_APPLICATION_CREDENTIALS }}
\ No newline at end of file
+        GCP_SA_KEY: ${{ secrets.FIREBASE_APPLICATION_CREDENTIALS }}
+        GITHUB_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
```
Comment (yuval-k) [inline .github/workflows/publish-docs.yaml:37]: is this needed?

File: projects/gloo/pkg/syncer/setup_syncer.go [hunk 2]
```diff
@@ -584,24 +579,6 @@ func RunGlooWithExtensions(opts bootstrap.Opts, extensions Extensions) error {
 		opts.ValidationServer.StartGrpcServer = false
 	}
 
-	go func() {
-		var handler metricsservice.MetricsHandler
-
-		if extensions.MetricsHandler == nil {
-			handler, err = metricsservice.NewConfigMapBackedDefaultHandler(opts.WatchOpts.Ctx)
```
Comment (yuval-k) [inline projects/gloo/pkg/syncer/setup_syncer.go:591]: can we / should we remove metricsservice package?

File: .github/workflows/publish-docs.yaml (hunk 1 above)
Comment (EItanya) [reply .github/workflows/publish-docs.yaml:37]: Yes, the script that generates the docs needs a `GITHUB_TOKEN` to generate the changelog. Also, this workflow only runs after a merge to master

File: projects/gloo/pkg/syncer/setup_syncer.go (hunk 2 above)
Comment (EItanya) [reply projects/gloo/pkg/syncer/setup_syncer.go:591]: I'd be fine with that, but I tend to err on side of deleting as little as possible

**Review by yuval-k — APPROVED** [review]

**Pull request merged**